components/match2/wikis/leagueoflegends/match_group_input_custom.lua

```diff
@@ -155,425 +151,255 @@ end
+	local firstTo = math.floor(match.bestof / 2)
+	if Array.any(opponents, function(opponent) return (tonumber(opponent.score) or 0) > firstTo end) then
+		return true
 	end
-
-	-- Remove special keys (maps/games, opponents, bracketdata etc)
-	for key, _ in pairs(standaloneMatch) do
-		if String.startsWith(key, 'match2') then
-			standaloneMatch[key] = nil
-		end
+	local scoreSum = Array.reduce(opponents, function(sum, opponent) return sum + (opponent.score or 0) end, 0)
+	if scoreSum >= match.bestof then
+		return true
 	end
 
-	-- Copy all match level records which have value
-	for key, value in pairs(standaloneMatch) do
-		if String.isNotEmpty(value) then
-			match[key] = value
-		end
+	-- If enough time has passed since match started, it should be marked as finished
+	local threshold = match.dateexact and SECONDS_UNTIL_FINISHED_EXACT
+		or SECONDS_UNTIL_FINISHED_NOT_EXACT
+	if match.timestamp ~= DateExt.defaultTimestamp and match.timestamp + threshold < NOW then
+		return true
 	end
 
-	return match
+	return false
 end
 
---
--- map related functions
---
-
 -- Parse extradata information
 ---@param MatchParser LeagueOfLegendsMatchParserInterface
 ---@param map table
 ---@return table
 function MapFunctions.getAdditionalExtraData(MatchParser, map)
-	map.extradata.comment = map.comment
-	map.extradata.team1side = MatchParser.getSide(map, 1) or ''
-	map.extradata.team2side = MatchParser.getSide(map, 2) or ''
-	map.extradata.team1objectives = MatchParser.getObjectives(map, 1) or {}
-	map.extradata.team2objectives = MatchParser.getObjectives(map, 2) or {}
-
-	return map
+	return {
+		comment = map.comment,
+		team1side = MatchParser.getSide(map, 1) or '',
+		team2side = MatchParser.getSide(map, 2) or '',
+		team1objectives = MatchParser.getObjectives(map, 1) or {},
+		team2objectives = MatchParser.getObjectives(map, 2) or {},
+	}
 end
 
 -- Parse participant information
 ---@param MatchParser LeagueOfLegendsMatchParserInterface
 ---@param map table
----@param opponents table[]
----@return table
-function MapFunctions.getParticipants(MatchParser, map, opponents)
+---@return table, table
+function MapFunctions.getParticipants(MatchParser, map)
 	local participants = {}
 	local extradata = {}
 	local getCharacterName = FnUtil.curry(MatchGroupInput.getCharacterName, HeroNames)
 
 	for opponentIndex = 1, MAX_NUM_OPPONENTS do
 		local teamPrefix = 'team' .. opponentIndex
+
 		Array.forEach(MatchParser.getHeroPicks(map, opponentIndex) or {}, function (hero, idx)
 			extradata[teamPrefix .. 'champion' .. idx] = getCharacterName(hero)
 		end)
 		Array.forEach(MatchParser.getHeroBans(map, opponentIndex) or {}, function (hero, idx)
 			extradata[teamPrefix .. 'ban' .. idx] = getCharacterName(hero)
 		end)
+
 		for playerIndex, participant in ipairs(MatchParser.getParticipants(map, opponentIndex) or {}) do
 			participant.character = getCharacterName(participant.character)
 			participants[opponentIndex .. '_' .. playerIndex] = participant
```

Comment:
not sure this works as intended
wouldn't we need a lookup to check for the correct playerIndex (match2player id within match2opponent)?
like the input order here doesn't need to match the order of players set by teamcard
and in that case we get the data attached to the wrong player

Reply:
On League it's fine, TC are in role order, manual inputs are in role order, and API input are being sorted in role order

Reply:
alrighty
when we come to other wikis we probably should make it a commons function though :)